pysatSpaceWeather/instruments/methods/ace.py

```diff
@@ -56,13 +61,232 @@ def references(iname):
             'swepam': ''.join(['McComas, D., Bame, S., Barker, P. et al. ',
                                'Solar Wind Electron Proton Alpha Monitor ',
                                '(SWEPAM) for the Advanced Composition ',
-                               'Explorer. Space Science Reviews 86, 563–612',
+                               'Explorer. Space Sci. Rev., 86, 563–612',
                                ' (1998). ',
                                'https://doi.org/10.1023/A:1005040232597']),
             'sis': ''.join(['Stone, E., Cohen, C., Cook, W. et al. The ',
                             'Solar Isotope Spectrometer for the Advanced ',
-                            'Composition Explorer. Space Science Reviews ',
-                            '86, 357–408 (1998). ',
+                            'Composition Explorer. Space Sci. Rev., 86, ',
+                            '357–408 (1998). ',
                             'https://doi.org/10.1023/A:1005027929871'])}
 
-    return refs[iname]
+    if name not in refs.keys():
+        raise KeyError('unknown ACE instrument, accepts {:}'.format(
+            refs.keys()))
+
+    return refs[name]
+
+
+def clean(inst):
+    """Common aspects of the ACE space weather data cleaning
+
+    Parameters
+    ----------
+    inst : pysat.Instrument
+        ACE pysat.Instrument object
+
+    Returns
+    -------
+    max_status : int
+        Maximum allowed status
+
+    Note
+    ----
+    pysat Instrument is modified in place
+
+    """
+
+    if inst.platform != "ace":
+        raise AttributeError("Can't apply ACE cleaning to platform {:}".format(
+            inst.platform))
+
+    # Set the clean level
+    if inst.clean_level == 'dusty':
+        logger.warning("unused clean level 'dusty', reverting to 'clean'")
+        inst.clean_level = 'clean'
+
+    max_status = 9
+    if inst.clean_level == "clean":
+        max_status = 0
+    elif inst.clean_level == "dirty":
+        max_status = 8
+
+    # Replace all fill values with NaN
```

Comment:
Ah, you already have the fill replacement code here. Note `clean` doesn't get called if clean_level is set to 'none' so the fill_val replacement is only sometimes. Perhaps this is as intended.

Reply:
It was, but I decided to change the behaviour and put that part in `preprocess`.